src/vcbuild/msvc-dmc.d

```diff
@@ -0,0 +1,62 @@
+/// Wrapper which accepts DMC command-line syntax
+/// and passes the transformed options to a MSVC cl.exe.
+module msvc_dmc;
+
+import std.algorithm.searching;
+import std.array;
+import std.file;
+import std.path;
+import std.process;
+import std.stdio;
+
+int main(string[] args)
+{
+    auto cl = environment.get("MSVC_CC",
+        environment.get("VCINSTALLDIR", `\Program Files (x86)\Microsoft Visual Studio 10.0\VC\`)
+            .buildPath("bin", "amd64", "cl.exe"));
+    string[] newArgs = [cl];
+    newArgs ~= `/Ivcbuild`;
+    newArgs ~= `/Iroot`;
+    newArgs ~= `/FIwarnings.h`;
+    bool compilingOnly;
+
+    foreach (arg; args[1..$])
+    {
+        switch (arg)
+        {
+            case "-cpp": // "source files are C++"
+                newArgs ~= "/TP";
+                break;
+            case "-e": // "show results of preprocessor"
+                break;
+            case "-gl": // "debug line numbers only"
+                newArgs ~= "/Zi";
+                break;
+            case "-wx": // "treat warnings as errors"
+                newArgs ~= "/WX";

```

Comment:
Wouldn't a better solution be to add it to `vcbuild\warnings.h`?

I'd like us to test building this configuration. The VS build historically broke a lot.